lib/ood_core/job/adapters/lsf.rb

```diff
@@ -0,0 +1,200 @@
+require "ood_core/refinements/hash_extensions"
+
+module OodCore
+  module Job
+    class Factory
+      using Refinements::HashExtensions
+
+      # Build the Lsf adapter from a configuration
+      # @param config [#to_h] the configuration for job adapter
+      # @option config [#to_s] :bin ('') Path to lsf client binaries
+      def self.build_lsf(config)
+        c = config.to_h.symbolize_keys
+        batch = Adapters::Lsf::Batch.new(bin: c.fetch(:bin, ""))
+        Adapters::Lsf.new(batch: batch)
+      end
+    end
+
+    module Adapters
+      class Lsf < Adapter
+        attr_reader :batch
+
+        require "ood_core/job/adapters/lsf/batch"
```

Comment:
No. If I do that, then the initial creation of the class Lsf occurs in the Lsf::Batch object.

Then I can't get away with doing this in Lsf::Batch:

```ruby
# lsf/batch.rb
class OodCore::Job::Adapters::Lsf::Batch
  # code here
end
```

I'd have to do:

```ruby
# lsf/batch.rb
module OodCore
  module Job
    module Adapters
      class Lsf < Adapter
        class Batch
          # code here
...
```

Though I could get away with this instead:

```ruby
# lsf/batch.rb
module OodCore
  module Job
    module Adapters
      class Lsf < Adapter
      end
    end
  end
end

class OodCore::Job::Adapters::Lsf::Batch
  # code here
end
```

It was easier to just move the require statement below.